pkg/autodiscovery/listeners/types.go

```diff
@@ -38,6 +38,7 @@ type Service interface {
 	IsReady() bool                             // is the service ready
 	GetCheckNames() []string                   // slice of check names defined in kubernetes annotations or docker labels
 	HasFilter(containers.FilterType) bool      // whether the service is excluded by metrics or logs exclusion config
+	GetSNMPInfo(string) (string, error)        // SNMP specific configuration values
```

Comment:
I don't think it is a good idea to some "SNMP" specificity in this interface. What you are introduce can be use in other use cases. So maybe we can rename this method like:
```suggestion
	GetExtraConfig(string) (string, error)        // retrieves specific Service configuration values.
```

Reply:
Ah that's what I thought but @ahmed-mez convinced me otherwise :). Is it fine to have then `extra_user` from the template? Or `extra_snmp_user` ?